hector_mapping/src/HectorMappingRos.cpp

```diff
@@ -256,9 +269,13 @@ void HectorMappingRos::scanCallback(const sensor_msgs::LaserScan& scan)
   {
     // If we are using the tf tree to find the transform between the base frame and laser frame,
     // let's get that transform
-    const ros::Duration dur(0.5);
     tf::StampedTransform laser_transform;
-    if (tf_.waitForTransform(p_base_frame_, scan.header.frame_id, scan.header.stamp, dur))
+    if (cached_static_lidar_transforms_.find(scan.header.frame_id) != cached_static_lidar_transforms_.end())
+    {
+      // Transform from base_frame to laser frame has been cached at node startup, so let's get it without a tf lookup
+      laser_transform = cached_static_lidar_transforms_[scan.header.frame_id];
+    }
+    else if (tf_.waitForTransform(p_base_frame_, scan.header.frame_id, scan.header.stamp, ros::Duration(0.5)))
```

Comment:
here we just check if the transform has been cached... if not, then look it up in the tf tree